metrics/node/subscribe.go

```diff
@@ -0,0 +1,47 @@
+package node
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/status-im/status-go/geth/log"
+)
+
+// SubscribeServerEvents subscribes to server and listens to
+// PeerEventTypeAdd and PeerEventTypeDrop events.
+func SubscribeServerEvents(ctx context.Context, node *node.Node) error {
```

Comment:
can we have unit tests for this function?

Reply:
Yes, we totally can! I will add them.